jvm/src/main/scala/kiama/util/Server.scala

```diff
@@ -495,6 +504,22 @@ trait LanguageService[N] {
 
   case class RelatedInfo(range: Option[Range], message: String)
 
+  enum InlayHintKind {
+    case Type, Parameter
+
+    def toLSP(): LSPInlayHintKind = this match
+      case InlayHintKind.Type => LSPInlayHintKind.Type
+      case InlayHintKind.Parameter => LSPInlayHintKind.Parameter
+  }
+
+  /**
+   * A representation of a simple inlay hint.
+   *
+   * TODO: Support for attached text edit[s] (that must materialize the inlay hint, making it obsolete).
```

Comment:
- it would be nice to "just" have a simple 'yep, you can just insert the label as code and it just works' for types
- we should also hijack the `data` field in LSP to send what exact kind of an inlay it is: for Effekt, we might have a bunch of them and the client should be able to filter things out somehow (and the standard only allows a kind `Type` and `Parameter`, lol 🙃)

Reply:
both are done :)